backend/deepchecks_monitoring/api/v1/data_ingestion_alert_rule.py

```diff
@@ -0,0 +1,177 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+"""V1 API of the alert rules."""
+import typing as t
+
+from fastapi import Depends, Response, status
+from pydantic import BaseModel
+from sqlalchemy import insert, select, update
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from deepchecks_monitoring.api.v1.data_ingestion_alert import DataIngestionAlertSchema
+from deepchecks_monitoring.config import Tags
+from deepchecks_monitoring.dependencies import AsyncSessionDep
+from deepchecks_monitoring.monitoring_utils import IdResponse
+from deepchecks_monitoring.public_models.user import User
+from deepchecks_monitoring.schema_models.data_ingestion_alert import DataIngestionAlert
+from deepchecks_monitoring.schema_models.data_ingestion_alert_rule import AlertRuleType, DataIngestionAlertRule
+from deepchecks_monitoring.schema_models.model import Model
+from deepchecks_monitoring.utils import auth
+from deepchecks_monitoring.utils.alerts import AlertSeverity, Condition, Frequency
+
+from .router import router
+
+
+class DataIngestionAlertRuleCreationSchema(BaseModel):
+    """Schema defines the parameters for creating new alert rule."""
+
+    name: str
+    condition: Condition
+    alert_severity: t.Optional[AlertSeverity]
+    is_active: t.Optional[bool]
+    alert_type: AlertRuleType
+    frequency: Frequency
+
+
+class DataIngestionAlertRuleSchema(BaseModel):
+    """Schema for the alert rule."""
+
+    id: int
+    name: str
+    model_id: int
+    condition: Condition
+    alert_severity: t.Optional[AlertSeverity]
+    is_active: bool
+
+    class Config:
+        """Config for Alert schema."""
+
+        orm_mode = True
+
+
+class DataIngestionAlertRuleUpdateSchema(BaseModel):
+    """Schema defines the parameters for updating alert rule."""
+
+    name: t.Optional[str]
+    alert_severity: t.Optional[AlertSeverity]
+    condition: t.Optional[Condition]
+    is_active: t.Optional[bool]
+
+
+@router.post(
+    "/models/{model_id}/data-ingestion-alert-rules",
+    response_model=IdResponse,
+    tags=[Tags.ALERTS],
+    dependencies=[Depends(Model.get_object_from_http_request)],
```

Comment:
`Model.get_object_from_http_request`

I think we should decouple this into `Model.exists_or_404` and `Model.has_permission_or_403`/
I know that this is not related to this issue, so, maybe create an issue for this? 
of course, if you think that it has a sense